Tests/UnitTests/Plugins/Detray/DetrayMaterialConverterTests.cpp

```diff
@@ -0,0 +1,348 @@
+
+  std::vector<Acts::MaterialSlab> materialSlabs = {
+      materialSlab12345, materialSlab678910, materialSlab54321,
+      materialSlab109876};
+
+  auto binnedMaterial = Acts::BinnedSurfaceMaterial(
+      binUtility, {materialSlabs}, 0., Acts::MappingType::Default);
+
+  detray::io::grid_payload<detray::io::material_slab_payload,
+                           detray::io::material_id>
+      payload = Acts::DetrayMaterialConverter::convertSurfaceMaterial(
+          binnedMaterial, *logger);
+  // Check the payload
+  // - we fake 2D always for detray to minimize the number of containers
+  // - four material payloads in the grid
+  BOOST_CHECK_EQUAL(payload.axes.size(), 2u);
+  BOOST_CHECK_EQUAL(payload.bins.size(), 4u);
+  BOOST_CHECK(!payload.transform.has_value());
+  BOOST_CHECK(payload.grid_link.type ==
+              detray::io::material_id::rectangle2_map);
+  // x-axis is dummy
+  BOOST_CHECK(payload.axes[0u].label == detray::axis::label::e_x);
+  BOOST_CHECK_EQUAL(payload.axes[0u].bins, 1u);
+  // y-axis
+  BOOST_CHECK(payload.axes[1u].label == detray::axis::label::e_y);
+  BOOST_CHECK_EQUAL(payload.axes[1u].bins, 4u);
+  CHECK_CLOSE_ABS(payload.axes[1u].edges[0u], -2.,
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  CHECK_CLOSE_ABS(payload.axes[1u].edges[1u], 2.,
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  BOOST_CHECK(payload.axes[1u].binning == detray::axis::binning::e_regular);
+  BOOST_CHECK(payload.axes[1u].bounds == detray::axis::bounds::e_closed);
+}
+
+BOOST_AUTO_TEST_CASE(DetrayBinnedMaterialConversionXY) {
+  // Create a binned material in 2 x2  bins in x-y direction
+  Acts::BinUtility binUtility(2u, -1., 1., Acts::BinningOption::open,
+                              Acts::BinningValue::binX);
+  binUtility += Acts::BinUtility(2u, -2., 2., Acts::BinningOption::open,
+                                 Acts::BinningValue::binY);
+
+  std::vector<Acts::MaterialSlab> materialSlabs0 = {materialSlab12345,
+                                                    materialSlab678910};
+  std::vector<Acts::MaterialSlab> materialSlabs1 = {materialSlab54321,
+                                                    materialSlab109876};
+
+  auto binnedMaterial =
+      Acts::BinnedSurfaceMaterial(binUtility, {materialSlabs0, materialSlabs1},
+                                  0., Acts::MappingType::Default);
+
+  detray::io::grid_payload<detray::io::material_slab_payload,
+                           detray::io::material_id>
+      payload = Acts::DetrayMaterialConverter::convertSurfaceMaterial(
+          binnedMaterial, *logger);
+  // Check the payload
+  BOOST_CHECK_EQUAL(payload.axes.size(), 2u);
+  BOOST_CHECK(payload.grid_link.type ==
+              detray::io::material_id::rectangle2_map);
+  //  The axis are real x-y
+  BOOST_CHECK(payload.axes[0u].label == detray::axis::label::e_x);
+  BOOST_CHECK(payload.axes[1u].label == detray::axis::label::e_y);
```

Comment:
Can we also check the number of bins for 2x2 binned material?

Reply:
Done.